fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AggregateDisassemble.java

```diff
@@ -0,0 +1,151 @@
+                    List<Expression> expressionList = functionCall.getArguments();
+                    List<Type> staleTypeList = expressionList.stream().map(Expression::getDataType)
+                            .map(DataType::toCatalogDataType).collect(Collectors.toList());
+                    Function staleFuncDesc = new Function(functionName, staleTypeList,
+                            functionCall.getDataType().toCatalogDataType(),
+                            // I think an aggregate function will never have a variable length parameters
+                            false);
+                    Function staleFunc = Catalog.getCurrentCatalog()
+                            .getFunction(staleFuncDesc, CompareMode.IS_IDENTICAL);
+                    Preconditions.checkArgument(staleFunc instanceof org.apache.doris.catalog.AggregateFunction);
+                    org.apache.doris.catalog.AggregateFunction
+                            staleAggFunc = (org.apache.doris.catalog.AggregateFunction) staleFunc;
+                    Type staleIntermediateType = staleAggFunc.getIntermediateType();
+                    Type staleRetType = staleAggFunc.getReturnType();
+                    if (staleIntermediateType != null && !staleIntermediateType.equals(staleRetType)) {
+                        functionCall.setIntermediate(DataType.convertFromCatalogDataType(staleIntermediateType));
+                    }
+                }
+                intermediateAggExpressionList.add(namedExpression);
+            }
+            LogicalAggregate localAgg = new LogicalAggregate(
+                    agg.getGroupByExprList().stream().map(Expression::clone).collect(Collectors.toList()),
+                    intermediateAggExpressionList,
+                    true
+            );
+
+            Plan childPlan = plan(localAgg, plan.child(0));
+            List<Slot> stalePlanOutputSlotList = plan.getOutput();
+            List<Slot> childOutputSlotList = childPlan.getOutput();
+            int childOutputSize = stalePlanOutputSlotList.size();
+            Preconditions.checkState(childOutputSize == childOutputSlotList.size());
+            Map<Slot, Slot> staleToNew = new HashMap<>();
+            for (int i = 0; i < stalePlanOutputSlotList.size(); i++) {
+                staleToNew.put(stalePlanOutputSlotList.get(i), childOutputSlotList.get(i));
+            }
+            List<Expression> groupByExpressionList = agg.getGroupByExprList();
+            for (int i = 0; i < groupByExpressionList.size(); i++) {
+                replaceSlot(staleToNew, groupByExpressionList, groupByExpressionList.get(i), i);
+            }
+            List<NamedExpression> mergeOutputExpressionList = agg.getOutputExpressionList();
+            for (int i = 0; i < mergeOutputExpressionList.size(); i++) {
+                replaceSlot(staleToNew, mergeOutputExpressionList, mergeOutputExpressionList.get(i), i);
+            }
+            LogicalAggregate mergeAgg = new LogicalAggregate(
```

Comment:
You need a representation to determine whether the current agg is a lower agg or an upper one.

Reply:
done